src/snowflake/connector/s3_storage_client.py

```diff
@@ -197,48 +263,82 @@ def _send_request_with_authentication_and_retry(
         self,
         url: str,
         verb: str,
-        resources: str,
         retry_id: Union[int, str],
+        query_parts: Optional[Dict[str, str]] = None,
         x_amz_headers: Optional[Dict[str, str]] = None,
         headers: Optional[Dict[str, str]] = None,
-        content_type: str = "",
-        data: Union[bytes, bytearray, IOBase, None] = None,
+        payload: Union[bytes, bytearray, IOBase, None] = None,
+        unsigned_payload: bool = False,
```

Comment:
Here I'm adding this new ability to do an unsigned payload when doing a HTTP request. This was previously not a thing with V2 signatures.

Reply:
Do you see us making `HTTP` requests at all?

Reply:
I thought everything was `HTTPS`